--- src/mvrtree/MVRTree.h.orig	2007-07-08 12:56:25.000000000 +0100
+++ src/mvrtree/MVRTree.h	2007-07-08 12:57:22.000000000 +0100
@@ -32,8 +32,32 @@
 	{
 		class MVRTree : public ISpatialIndex
 		{
-			class NNEntry;
-			class RootEntry;
+			class RootEntry
+			{
+			public:
+				RootEntry() {}
+				RootEntry(id_type id, double s, double e) : m_id(id), m_startTime(s), m_endTime(e) {}
+
+				id_type m_id;
+				double m_startTime;
+				double m_endTime;
+			}; // RootEntry
+
+			class NNEntry
+			{
+			public:
+				id_type m_id;
+				IEntry* m_pEntry;
+				double m_minDist;
+
+				NNEntry(id_type id, IEntry* e, double f) : m_id(id), m_pEntry(e), m_minDist(f) {}
+				~NNEntry() {}
+
+				struct greater : public std::binary_function<NNEntry*, NNEntry*, bool>
+				{
+					bool operator()(const NNEntry* __x, const NNEntry* __y) const { return __x->m_minDist > __y->m_minDist; }
+				};
+			}; // NNEntry
 
 		public:
 			MVRTree(IStorageManager&, Tools::PropertySet&);
@@ -155,32 +179,6 @@
 			bool m_rwLock;
 #endif
 
-			class RootEntry
-			{
-			public:
-				RootEntry() {}
-				RootEntry(id_type id, double s, double e) : m_id(id), m_startTime(s), m_endTime(e) {}
-
-				id_type m_id;
-				double m_startTime;
-				double m_endTime;
-			}; // RootEntry
-
-			class NNEntry
-			{
-			public:
-				id_type m_id;
-				IEntry* m_pEntry;
-				double m_minDist;
-
-				NNEntry(id_type id, IEntry* e, double f) : m_id(id), m_pEntry(e), m_minDist(f) {}
-				~NNEntry() {}
-
-				struct greater : public std::binary_function<NNEntry*, NNEntry*, bool>
-				{
-					bool operator()(const NNEntry* __x, const NNEntry* __y) const { return __x->m_minDist > __y->m_minDist; }
-				};
-			}; // NNEntry
 
 			class NNComparator : public INearestNeighborComparator
 			{
